draft-ietf-quic-transport.md

```diff
@@ -3838,143 +3838,135 @@ The details of loss detection and congestion control are described in
-server's preferred address, and on active connection migration to a new path.
-{{ecn-alg}} describes one possible algorithm for testing paths for ECN support.
+An ECN count is not incremented when the received IP packet does not results in
+a QUIC packets being processed.  A QUIC packet detected by a receiver as a
+duplicate also does not increase an ECN count; see {{security-ecn}} for relevant
+security concerns.
 
-Even if an endpoint does not use ECN markings on packets it transmits, the
-endpoint MUST provide feedback about ECN markings received from the peer if they
-are accessible.  Failing to report ECN counts will cause the peer to disable ECN
-marking.
 
+### ECN Validation {#ecn-validation}
 
-#### Sending ECN Markings
+It is possible for faulty network devices to corrupt or erroneously drop packets
+that set an ECN codepoint.  To provide robust connectivity in the presence of
+such devices, an endpoint validates the ECN counts for each network path  and
+disables use of ECN on that path if errors are detected.
 
-To start ECN validation, an endpoint SHOULD do the following when sending
-packets on a new path to a peer:
+To perform ECN validation for a new path:
 
-* Set the ECT(0) codepoint in the IP header of early outgoing packets sent on a
-  new path to the peer ({{!RFC8311}}).
+* The endpoint SHOULD set the ECT(0) codepoint in the IP header of early
+  outgoing packets sent on a new path to the peer ({{!RFC8311}}).
 
-* If all packets that were sent with the ECT(0) codepoint are eventually deemed
-  lost (Section 6 of {{QUIC-RECOVERY}}), validation is deemed to have failed.
+* The endpoint SHOULD monitor whether all packets sent with an ECT codepoint are
+  eventually deemed lost (Section 6 of {{QUIC-RECOVERY}}), deeming this as an
+  indication that ECN validation has failed.
 
-To reduce the chances of misinterpreting congestive loss as packets dropped by a
-faulty network element, an endpoint could set the ECT(0) codepoint for only the
-first ten outgoing packets on a path, or for a period of three RTTs, whichever
-occurs first.
+To reduce the chances of misinterpreting loss of packets dropped by a faulty
+network element, an endpoint could set an ECT codepoint for only the first ten
+outgoing packets on a path, or for a period of three RTTs, whichever occurs
+first.
+
+An endpoint thus attempts to use ECN and validates this for each new connection,
+when switching to a server's preferred address, and on active connection
+migration to a new path.  {{ecn-alg}} describes one possible algorithm.
 
 Other methods of probing paths for ECN support are possible, as are different
 marking strategies. Implementations MAY use other methods defined in RFCs; see
-{{?RFC8311}}. Implementations that use the ECT(1) codepoint need to perform ECN
-validation using ECT(1) counts.
+{{?RFC8311}}. Implementations that use the ECT(1) codepoint need to
+perform ECN validation using the reported ECT(1) counts.
```

Comment:
This whole ECN text needs to make up its mind on whether it wants to specify that ECT(0) should be used, or if it wants to allow ECT(0) or ECT(1). At the moment, it's a mix. And if it wants to allow both, some words need to be added whether an implementation should pick one statically or randomly or whatnot.